src/module.c

```diff
@@ -221,27 +248,34 @@ static void sentbox_thread_func(void* arg)
 
 static void finalize_chat(napi_env env, void* data, void* hint) {
   if (data) {
-    dc_chat_unref((dc_chat_t*)data);
+    dc_chat_t* chat = (dc_chat_t*)data;
+    //TRACE("cleaning up chat %d", dc_chat_get_id(chat));
+    dc_chat_unref(chat);
   }
 }
 
 static void finalize_chatlist(napi_env env, void* data, void* hint) {
   if (data) {
+    //TRACE("cleaning up chatlist object");
     dc_chatlist_unref((dc_chatlist_t*)data);
   }
 }
 
 static void finalize_contact(napi_env env, void* data, void* hint) {
   if (data) {
-    dc_contact_unref((dc_contact_t*)data);
+    dc_contact_t* contact = (dc_contact_t*)data;
+    //TRACE("cleaning up contact %d", dc_contact_get_id(contact));
+    dc_contact_unref(contact);
   }
 }
 
 static void finalize_context(napi_env env, void* data, void* hint) {
   if (data) {
     dcn_context_t* dcn_context = (dcn_context_t*)data;
     dcn_context->gc = 1;
+    TRACE("cleaning up dc_context..");
     dc_context_unref(dcn_context->dc_context);
```

Comment:
@dignifiedquire This is the call that hangs and takes a long time to finish. Since `finalize_context()` is called as a result of v8 garbage collection, this will make the render thread hang in electron.

Reply:
Good find! Any ideas how to fix? I looked into this already in kiev, but didn't get that far with investigating. The interrupt_mvbox thing also blocks on c core.